openedx/core/djangoapps/user_api/partition_schemes.py

```diff
@@ -10,6 +10,27 @@
 log = logging.getLogger(__name__)
 
 
+class NotImplementedPartitionScheme(object):
```

Comment:
I like this general idea! I'm not sure though about key_for_partition. Why is that needed?